### PR TITLE
TypeChoiceDef to Choice module

### PR DESCRIPTION
**Pull request by adjivas**
Hello @ulysseB.

This PR is a next reorganization of Context module for IR Set implementation after [TypeSetDef to Set module](https://github.com/ulysseB/telamon/pull/115).
This PR imply:
* move register methods from [context](https://github.com/ulysseB/telamon/pull/127/files#diff-d0966ad8d32495f16a95bcb4b576d81f) module to [enumeration](https://github.com/ulysseB/telamon/pull/127/files#diff-6c689d52a08f7be368551ce2df87cf5a) module.
* move register methods from [context](https://github.com/ulysseB/telamon/pull/127/files#diff-d0966ad8d32495f16a95bcb4b576d81f) module to [integer](https://github.com/ulysseB/telamon/pull/127/files#diff-d22020b5da55c34e20d6f329f4cc9ae6) module.
* prepare a [counter](https://github.com/ulysseB/telamon/pull/127/files#diff-064c9253a5deb445cd8048374a37c89f) submodule from [choice](https://github.com/ulysseB/telamon/pull/127/files#diff-424eaf4bd4ab66adb24b21dde8288a56) module.

The next PR will move CounterDef to submodule Counter from [choice](https://github.com/ulysseB/telamon/pull/127/files#diff-424eaf4bd4ab66adb24b21dde8288a56) module.

There aren't particular changes.

### CODE REVIEW

**Comment by adjivas** [comment]
bors try

File: telamon-gen/src/ast/choice/integer.rs [hunk 1]
```diff
@@ -33,13 +33,42 @@ impl IntegerDef {
...
     pub fn define(
         self,
         context: &mut CheckerContext,
         tc: &mut TypingContext,
     ) -> Result<(), TypeError> {
         self.check_undefined_variables(context)?;
+
+        self.define_integer(tc);
+
+        tc.choice_defs.push(ChoiceDef::IntegerDef(self));
```
[the remark this answers is not in the document]
Comment (adjivas) [reply telamon-gen/src/ast/choice/integer.rs]: The idea is to remove the [add_statement](https://github.com/ulysseB/telamon/blob/49ee04502e2ac9ff4c8ad738ce191dcc5da4df98/telamon-gen/src/ast/context.rs#L16) method from Typing Context module.
I have notified a [TODO](https://github.com/ulysseB/telamon/blob/49ee04502e2ac9ff4c8ad738ce191dcc5da4df98/telamon-gen/src/ast/mod.rs#L42) where the instruction is to remove the Typing Context

[the remark this answers is not in the document]
Comment (ulysseB) [reply telamon-gen/src/ast/choice/integer.rs]: Ok.

[the remark this answers is not in the document]
Comment (adjivas) [reply telamon-gen/src/ast/choice/integer.rs]: @ulysseB That's what you expect from me, right?

File: telamon-gen/src/ast/choice/enumeration.rs [hunk 2]
```diff
@@ -245,6 +367,10 @@ impl EnumDef {
         self.check_two_parameter()?;
         self.check_same_parameter()?;
         self.check_conflict()?;
+
+        self.register_enum(tc);
+
+        tc.choice_defs.push(ChoiceDef::EnumDef(self));
```
[the remark this answers is not in the document]
Comment (adjivas) [reply telamon-gen/src/ast/choice/enumeration.rs]: same

File: telamon-gen/src/ast/choice/integer.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (ulysseB) [reply telamon-gen/src/ast/choice/integer.rs]: Yes, I you have the time to do it.

[the remark this answers is not in the document]
Comment (ulysseB) [reply telamon-gen/src/ast/choice/integer.rs]: Otherwise, it's not a problem.

[the remark this answers is not in the document]
Comment (adjivas) [reply telamon-gen/src/ast/choice/integer.rs]: That my objective :cat:

**Pull request merged**